Source/AssetRipper.Processing/AnimatorControllers/AnimatorStateContext.cs

```diff
@@ -0,0 +1,190 @@
+using AssetRipper.Assets.Collections;
+using AssetRipper.Assets.Generics;
+using AssetRipper.SourceGenerated.Classes.ClassID_1102;
+using AssetRipper.SourceGenerated.Classes.ClassID_91;
+using AssetRipper.SourceGenerated.Extensions;
+using AssetRipper.SourceGenerated.Subclasses.StateConstant;
+using AssetRipper.SourceGenerated.Subclasses.StateMachineConstant;
+
+namespace AssetRipper.Processing.AnimatorControllers;
+
+public sealed class AnimatorStateContext
+{
+	public readonly int StateCount;
+	public bool HasStates() => StateCount > 0;
+	public readonly int DefaultStateIdx;
+
+	private readonly ProcessedAssetCollection VirtualFile;
+	private readonly IAnimatorController Controller;
+	private readonly IStateMachineConstant StateMachineConstant;
+	private readonly int LayerIndex;
+	
+	private readonly IAnimatorState[] states;
+	private readonly IStateConstant[] stateConstants;
+	private readonly uint[] stateIdxsToStateMachinePathIDs; // for grouping States into StateMachines
```

Comment:
`stateIndexToStateMachinePathID` and please make the comment into xml documentation.